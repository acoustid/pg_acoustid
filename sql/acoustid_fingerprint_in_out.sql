CREATE EXTENSION acoustid;

SELECT '13123@$#@!'::acoustid_fingerprint;

SELECT 'AQAAO4qkJVGqKIH_An_hwx1-ofiLHj9-_NA1AvUT8K5x4sel4JGQrEda4YenG940o8h_JM4aw5_BH_dy_GiUFbW0oM2I5sgevHgLzUfjJEdJw6_Rj2De5PCHWjj64OLR5zjxo60M18I5wSt-AAWcMVA4JR4SxmigBEOOKAEFIIwIJoRgSgiLAAUEKAEGEkIRCwgCAA'::acoustid_fingerprint;
