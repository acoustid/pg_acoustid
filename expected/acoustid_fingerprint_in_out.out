CREATE EXTENSION IF NOT EXISTS acoustid;
SELECT '13123@$#@!'::acoustid_fingerprint;
ERROR:  invalid base64 string
LINE 1: SELECT '13123@$#@!'::acoustid_fingerprint;
               ^
SELECT 'AQAAO4qkJVGqKIH_An_hwx1-ofiLHj9-_NA1AvUT8K5x4sel4JGQrEda4YenG940o8h_JM4aw5_BH_dy_GiUFbW0oM2I5sgevHgLzUfjJEdJw6_Rj2De5PCHWjj64OLR5zjxo60M18I5wSt-AAWcMVA4JR4SxmigBEOOKAEFIIwIJoRgSgiLAAUEKAEGEkIRCwgCAA'::acoustid_fingerprint;
                                                                                          acoustid_fingerprint                                                                                          
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 AQAAO4qkJVGqKIH_An_hwx1-ofiLHj9-_NA1AvUT8K5x4sel4JGQrEda4YenG940o8h_JM4aw5_BH_dy_GiUFbW0oM2I5sgevHgLzUfjJEdJw6_Rj2De5PCHWjj64OLR5zjxo60M18I5wSt-AAWcMVA4JR4SxmigBEOOKAEFIIwIJoRgSgiLAAUEKAEGEkIRCwgCAA
(1 row)

SELECT CAST('AQAAE4qkJVGqKIH_An_hwx1-ofiLHj9-_NA1AvUT8K5x4gcFnDFQOCUeEsZooARDDg'::acoustid_fingerprint AS bytea);
                                                bytea                                                 
------------------------------------------------------------------------------------------------------
 \x010000138aa42551aa2881ff027fe1c31d7ea1f88b1e3f7efcd03502f513f0ae71e207059c315038251e12c668a004430e
(1 row)

SELECT CAST('AQAAE4qkJVGqKIH_An_hwx1-ofiLHj9-_NA1AvUT8K5x4gcFnDFQOCUeEsZooARDDg'::acoustid_fingerprint AS text);
                                text                                
--------------------------------------------------------------------
 AQAAE4qkJVGqKIH_An_hwx1-ofiLHj9-_NA1AvUT8K5x4gcFnDFQOCUeEsZooARDDg
(1 row)

SELECT extract_version('AQAAE4qkJVGqKIH_An_hwx1-ofiLHj9-_NA1AvUT8K5x4gcFnDFQOCUeEsZooARDDg'::acoustid_fingerprint);
 extract_version 
-----------------
               1
(1 row)

SELECT extract_terms('AQAAE4qkJVGqKIH_An_hwx1-ofiLHj9-_NA1AvUT8K5x4gcFnDFQOCUeEsZooARDDg'::acoustid_fingerprint);
                                                                                               extract_terms                                                                                                
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 {994277718,440498550,440498550,180443510,180476246,180485494,193052022,193052006,1233231206,1501666678,1484897654,1485028854,2021900278,2021759732,2021759732,2034343652,1761842924,1761773292,1744995052}
(1 row)

SELECT uncompress('AQAAE4qkJVGqKIH_An_hwx1-ofiLHj9-_NA1AvUT8K5x4gcFnDFQOCUeEsZooARDDg'::acoustid_fingerprint);
                                                                                                 uncompress                                                                                                 
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 {994277718,440498550,440498550,180443510,180476246,180485494,193052022,193052006,1233231206,1501666678,1484897654,1485028854,2021900278,2021759732,2021759732,2034343652,1761842924,1761773292,1744995052}
(1 row)

SELECT CAST('AQAAE4qkJVGqKIH_An_hwx1-ofiLHj9-_NA1AvUT8K5x4gcFnDFQOCUeEsZooARDDg'::acoustid_fingerprint AS acoustid_raw_fingerprint);
                                                                                          acoustid_raw_fingerprint                                                                                          
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 {994277718,440498550,440498550,180443510,180476246,180485494,193052022,193052006,1233231206,1501666678,1484897654,1485028854,2021900278,2021759732,2021759732,2034343652,1761842924,1761773292,1744995052}
(1 row)

